torch/csrc/jit/passes/freeze_module.cpp

```diff
@@ -218,10 +190,13 @@ class AttributePropagator {
         attr = IValue(t);
       }
     } else if (attr.isTuple()) {
-      std::vector<IValue>& elems = attr.toTuple()->elements();
+      auto tuple = std::move(attr).toTuple();
+      std::vector<IValue>& elems = tuple->elements();
       for (auto& elem : elems) {
         elem = overrideGradient(elem);
       }
+      attr = std::move(tuple);
+
     } else if (attr.isList()) {
       c10::List<IValue> elems = std::move(attr).toList();
```

Comment:
When we iterate over the dict here we're iterating as:
```
      for (const auto& pair : dict) {
        auto val = pair.value();
        val = overrideGradient(val);
      }
```

val here is a const pointer right so what does setting `val = overrideGradient(val);` do?

in general i dont really understand how this function works. with `Tensor` we are constructing a new tensor and returning it, list in place modifying, tuple and dict constructing or in-place modifying.

Reply:
Added std::move semantic so it is not in-place